### PR TITLE
opencv plugin

### PR DESCRIPTION
**Pull request by piiswrong**
@winstywang

### CODE REVIEW

File: plugin/opencv/cv_api.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#define PLUGIN_OPENCV_CV_API_H_
+
+#include <mxnet/c_api.h>
+
+MXNET_DLL int MXCVImdecode(
+  const unsigned char *img,
+  const mx_uint len,
+  const int flag,
+  NDArrayHandle *out);
+
```
Comment (tqchen) [inline plugin/opencv/cv_api.h]: can we make the API registerable as mxnet's ndarray API? So most cases there is no need in change of API interface and python

**Comment by tqchen** [comment]
I think it would be great if we only allow very few of API to appear on C API side, while making most of them registrable. So no change in client code is needed when adding more features

**Comment by piiswrong** [comment]
need to parse to/from string that way

**Comment by tqchen** [comment]
Here is the thing. I can find a way to solve the string parsing problem for symbolic Ops. So as long as you bind your execution pipeline as executor, there is no string parsing overhead.

Let us simply make these functions as functions in operator_util.h

**Comment by piiswrong** [comment]
That's not possible because each time the parameters are different due to random size and cropping

**Comment by tqchen** [comment]
hmm. I still think it is better to keep API concise as it is. It is likely that the string parsing speed won't be too much bottleneck anyway.

**Comment by piiswrong** [comment]
I'll merge this first. We can move the interface later

**Pull request merged**